graspologic/inference/latent_distribution_test.py

```diff
@@ -176,317 +189,231 @@ class LatentDistributionTest(BaseInference):
-        X_sampled = np.zeros(X.shape)
-        # TODO may be parallelized, but requires keeping track of random state
-        for i in range(N):
-            X_sampled[i, :] = X[i, :] + stats.multivariate_normal.rvs(cov=X_sigmas[i])
-
-        # return the embeddings in the appropriate order
-        return (Y, X_sampled) if reverse_order else (X_sampled, Y)
-
-    def fit(self, A1, A2):
-        """
-        Fits the test to the two input graphs
-
-        Parameters
-        ----------
-        A1, A2 : variable (see description)
-            The two graphs, or their embeddings to run a hypothesis test on.
-            Expected variable type and shape depends on input_graph attribute:
-
-            - input_graph=True
-                expects two unembedded graphs either as NetworkX graph objects, or as
-                two np.ndarrays, representing the adjacency matrices. In this
-                case will be embedded using adjacency spectral embedding.
-            - input_graph-False
-                expects two already embedded graphs. In this case they must be
-                arrays of shape (n, d) and (m, d), where d, the number of
-                components, must be shared.
-
-            Note that regardless of how the graphs are passed, they need not
-            have the same number of vertices.
-
-        Returns
-        -------
-        self
-
-        """
-        if self.input_graph:
-            A1 = import_graph(A1)
-            A2 = import_graph(A2)
-
-            X1_hat, X2_hat = self._embed(A1, A2)
-        else:
-            # check for nx objects, since they are castable to arrays,
-            # but we don't want that
-            if not isinstance(A1, np.ndarray):
-                msg = (
-                    f"Embedding of the first graph is of type {type(A1)}, not "
-                    "np.ndarray. If input_graph is False, the inputs need to be "
-                    "adjacency spectral embeddings, with shapes (n, d) and "
-                    "(m, d), passed as np.ndarrays."
-                )
-                raise TypeError(msg)
-            if not isinstance(A2, np.ndarray):
-                msg = (
-                    f"Embedding of the second graph is of type {type(A2)}, not an "
-                    "array. If input_graph is False, the inputs need to be "
-                    "adjacency spectral embeddings, with shapes (n, d) and "
-                    "(m, d), passed as np.ndarrays."
-                )
-                raise TypeError(msg)
+    metric_func_ = _instantiate_metric_func(metric=metric, test=test)
+    test_obj = KSample(test, compute_distance=metric_func_)
+
+    data = test_obj.test(X1_hat, X2_hat, reps=n_bootstraps, workers=workers, auto=False)
+
+    null_distribution_ = test_obj.indep_test.null_dist
+    sample_T_statistic_ = data[0]
+    p_value_ = data[1]
 
-            if A1.ndim != 2:
+    return p_value_, sample_T_statistic_, null_distribution_
```

Comment:
i thought we like named tuples in graspologic now, but i might be wrong on this one.

Reply:
@bdpedigo do you suggest I return named tuples for lpt and ldt?